### PR TITLE
add devise

### PR DESCRIPTION
**Pull request by camilevahviraki**
### **In this branch we did the following:**

- Install and setup devise.
- User can:
   - Can register a new user.
   - User logs in with a combination of email and password.
   - Hashed password should be stored in the database.
   - Ask for confirmation of email.
   - Can reset password.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,10 +1,13 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (Davidosky007) [inline app/models/user.rb]: * Good job so far but you are yet to implement two requirements 

1. Your application Ask for confirmation of email
2. A user can reset the password in the application. when i tried resetting the password i got an error. below is a screenshot ⬇️ 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @camilevahviraki @Ogaga01 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
## Highlights
* No linter errors 👍 
* Well written PR description 👍 
* Good looking readme file 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by camilevahviraki** [comment]
Hello dear Reviewer,
The confirmation link will appear in the terminal.
Please consider that.

**Review by KDlamini — APPROVED** [review]
### Hi @camilevahviraki 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

## HIGHLIGHTS 🌟 

✔️ User authentication implemented correctly
✔️ Good PR description
✔️ Commit messages are meaningful
✔️ Correct use of GitFlow
✔️ No linter errors

# Status: Approved ✔️ 

simon

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by camilevahviraki** [comment]
Thank you @KDlamini

**Pull request merged**